src/components/CioQuiz/index.tsx

```diff
@@ -1,20 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import QuizContext, { QuizContextValue } from './context';
 import QuizQuestions from '../QuizQuestions';
 import ResultContainer from '../ResultContainer/ResultContainer';
 import { RequestStates } from '../../constants';
 import Spinner from '../Spinner/Spinner';
 import useQuiz from '../../hooks/useQuiz';
+import SessionPromptModal from '../SessionPromptModal/SessionPromptModal';
 import { IQuizProps } from '../../types';
 
 export default function CioQuiz(props: IQuizProps) {
   const {
     cioClient,
     state,
-    events: { nextQuestion, previousQuestion, resetQuiz, addToCart, resultClick },
+    events: {
+      nextQuestion,
+      previousQuestion,
+      resetQuiz,
+      addToCart,
+      resultClick,
+      hydrateQuiz,
+      hasStoredState,
+      resetStoredState,
+    },
   } = useQuiz(props);
+  const [showSessionPrompt, setShowSessionPrompt] = useState(false);
+  const { resultsPageOptions, sessionStateOptions } = props;
 
-  const { resultsPageOptions } = props;
+  useEffect(() => {
+    if (sessionStateOptions?.showSessionModal !== undefined) {
```

Comment:
Can you add a comment here explaining this is to handle "default to true but respect the props if passed"?